GDJS/GDJS/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -1298,7 +1307,38 @@ gd::String EventsCodeGenerator::GenerateGetVariable(
     const gd::String& objectName) {
   gd::String output;
   const gd::VariablesContainer* variables = NULL;
-  if (scope == LAYOUT_VARIABLE) {
+  if (scope == ANY_VARIABLE) {
+    if (HasProjectAndLayout()) {
```

Comment:
For each `if (HasProjectAndLayout())` (here and below), can you add a `else { output = "// Unsupported local variables in functions."}` so that it's easy to find.